NSAttributedString+Markdown.m

```diff
@@ -541,6 +571,74 @@ static void updateAttributedString(NSMutableAttributedString *result, NSString *
+                                    checkCharacterRange = NSMakeRange(position, 1);
+                                }
+                                // verify character before bullet is either start of string or new line, otherwise could be a standard dash-space (e.g., "This - here - shouldn't create bullets")
+                                if (position > 0 && ![[scanString substringWithRange:NSMakeRange(position, 1)] isEqualToString:@"\n"]) {
+                                    // skip
+                                    replaceMarkers = NO;
+                                    break;
+                                }
+                            }
+                            NSUInteger bulletLevel = spacesBeforeBullet / 4;
+                            
+                            NSString *bulletCharacter = nil;
+                            switch (bulletLevel) {
+                                case 0:
+                                    bulletCharacter = bulletLevel1;
+                                    break;
+                                case 1:
+                                    bulletCharacter = bulletLevel2;
+                                    break;
+                                default:
+                                    bulletCharacter = bulletLevelGreaterThan2;
+                                    break;
+                            }
+                    
+                            UIFont *baseFont = [result attribute:NSFontAttributeName atIndex:(beginRange.location - mutationOffset) effectiveRange:NULL];
+                            UIFont *updatedBulletFont = [UIFont fontWithName:bulletFontName size:baseFont.pointSize];
+                            NSRange mutatedBeginRange = NSMakeRange(beginRange.location - mutationOffset - spacesBeforeBullet, beginRange.length + spacesBeforeBullet);
+                            [result replaceCharactersInRange:mutatedBeginRange withString:[NSString stringWithFormat:@"%@%@", bulletCharacter, @"\t"]];
+                            mutationOffset += spacesBeforeBullet;
+                            
+                            /*
+                             NSMutableAttributedString Bug? - apply bullet font adjustment AFTER replacing text or NSAttributedString will apply font
+                             change to beginning of list item text for mutationOffset # of characters
+                             */
+                            [result addAttribute:NSFontAttributeName value:updatedBulletFont range:NSMakeRange(mutatedBeginRange.location, 1)];
+                            
+                            NSRange listItemRangeWithBullet = NSMakeRange(mutatedBeginRange.location, endRange.location - mutatedBeginRange.location + endRange.length - mutationOffset - 1);
+                            NSMutableParagraphStyle *paragraphStyle = [[result attribute:NSParagraphStyleAttributeName atIndex:listItemRangeWithBullet.location effectiveRange:NULL] mutableCopy];
```

Comment:
Playing around with the markdown sample app built for macOS, sometimes paragraphStyle is nil, which causes an objective-c exception in the `[result addAttribute:value:range:]` call below. Perhaps it's a Mac-only issue.

I was able to hack around the exception by checking for nil paragraphStyle here and using `[[NSParagraphStyle defaultParagraphStyle] mutableCopy]` — not sure if that's the best solution though.

Anyway, in our ResearchKit code should we catch any objective-c exceptions when calling `initWithMarkdownRepresentation` to prevent crashes?